submission/persistence-mongo/src/main/kotlin/ac/uk/ebi/biostd/persistence/doc/db/data/SubmissionDraftDocDataRepository.kt

```diff
@@ -11,55 +11,75 @@ import ac.uk.ebi.biostd.persistence.doc.model.DocSubmissionDraft.Companion.USER_
 import ac.uk.ebi.biostd.persistence.doc.model.DocSubmissionDraft.DraftStatus
 import ac.uk.ebi.biostd.persistence.doc.model.DocSubmissionDraft.DraftStatus.ACCEPTED
 import ac.uk.ebi.biostd.persistence.doc.model.DocSubmissionDraft.DraftStatus.ACTIVE
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.reactive.asFlow
+import kotlinx.coroutines.reactor.awaitSingle
+import kotlinx.coroutines.reactor.awaitSingleOrNull
 import org.springframework.data.domain.PageRequest
-import org.springframework.data.mongodb.core.MongoTemplate
+import org.springframework.data.mongodb.core.ReactiveMongoTemplate
 import org.springframework.data.mongodb.core.query.Criteria.where
 import org.springframework.data.mongodb.core.query.Query
 import org.springframework.data.mongodb.core.query.Update.update
 
 class SubmissionDraftDocDataRepository(
     private val submissionDraftRepository: SubmissionDraftRepository,
-    private val mongoTemplate: MongoTemplate,
+    private val mongoTemplate: ReactiveMongoTemplate,
 ) : SubmissionDraftRepository by submissionDraftRepository {
-    fun saveDraft(userId: String, key: String, content: String): DocSubmissionDraft {
+    suspend fun saveDraft(userId: String, key: String, content: String): DocSubmissionDraft {
         val draft = DocSubmissionDraft(userId, key, content, ACTIVE)
         return mongoTemplate.replaceOrCreate(
             Query(where(USER_ID).`is`(userId).andOperator(where(KEY).`is`(key), where(STATUS).`is`(ACTIVE))),
             draft
         )
     }
 
-    fun setStatus(userEmail: String, key: String, newStatus: DraftStatus) {
+    fun findAllDraft(): Flow<DocSubmissionDraft> {
```

Comment:
I know I'm being picky here but, the name should be `findAllDrafts` (plural)

Reply:
because this class implement spring repository interface and we can not override a method and change return type (to use kotlin supsend and Flow instead of mono and Flux)